tests/ballerina-unit-test/src/test/resources/test-src/expressions/builtinoperations/length-operation.bal

```diff
@@ -0,0 +1,258 @@
+    int[] arr = [];
+    arr[0] = x;
+    arr[1] = y;
+    arr[2] = arr[0] + arr[1];
+    return arr.length();
+}
+
+function arrayLengthAccessTestMultiReturnStatementCase (int x, int y) returns (int,int,int) {
+    int[] arr = [];
+    arr[0] = x;
+    arr[1] = y;
+    arr[2] = arr[0] + arr[1];
+    int[] brr = [];
+    brr[0] = 1;
+    int[] crr = [];
+    crr[0] = 1;
+    crr[1] = x + y;
+    return (arr.length(), brr.length(), crr.length());
+}
+
+function arrayLengthAccessTestTypeCastExpressionCase (int x, int y) returns (int) {
+    int[] arr = [];
+    arr[0] = x;
+    arr[1] = y;
+    arr[2] = arr[0] + arr[1];
+    int lengthVal = <int> arr.length();
+    return lengthVal;
+}
+
+function arrayLengthAccessTestIfConditionCase (int x, int y) returns (int) {
+    int[] arr = [];
+    arr[0] = x;
+    arr[1] = y;
+    arr[2] = arr[0] + arr[1];
+    if (arr.length() == 3) {
+        return 3;
+    } else{
+        return 0;
+    }
+}
+
+function arrayLengthAccessTestBinaryExpressionCase (int x, int y) returns (int) {
+    int[] arr = [];
+    arr[0] = x;
+    arr[1] = y;
+    arr[2] = arr[0] + arr[1];
+    if (arr.length() == arr.length()) {
+        return 3;
+    } else {
+        return 0;
+    }
+}
+
+function arrayLengthAccessTestStructFieldAccessCase (int x, int y) returns (int) {
+    int[] arr = [];
+    arr[0] = x;
+    arr[1] = y;
+    arr[2] = arr[0] + arr[1];
+    Person jack = {name:"Jack", days:arr};
+
+    if (jack.days.length() == 3) {
+        return 3;
+    } else {
+        return 0;
+    }
+}
+
+type Person record {
+    string name;
+    int[] days;
+};
+
+type Employee record {
+    int id;
+    string name;
+    float salary;
+};
+
+type Empty record {};
+
+function arrayLengthAccessTestJSONArrayCase (int x, int y) returns (int) {
+    json arr = [x,y,5,5,6,6];
+    int lengthVal;
+    lengthVal = arr.length();
+    return lengthVal;
+}
+
+function lengthOfMap () returns (int) {
+    map namesMap = {fname:"foo", lname:"bar", sname:"abc", tname:"pqr"};
+    int lengthVal = namesMap.length();
+    return lengthVal;
+}
+
+function lengthOfEmptyMap () returns (int) {
+    map namesMap;
+    int lengthVal = namesMap.length();
+    return lengthVal;
+}
+
+function lengthOfSingleXmlElement() returns (int) {
+    xml x1 = xml `<book>The Lost World</book>`;
+    int lengthVal = x1.length();
+    return lengthVal;
+}
+
+function lengthOfMultipleXmlElements() returns (int) {
+    xml x1 = xml `<book>The Lost World</book>`;
+    xml x2 = xml `Hello, world!`;
+    xml x3 = xml `<!--I am a comment-->`;
+    xml x4 = xml `<?target data?>`;
+    xml x5 = x1 + x2 + x3 + x4;
+    int lengthVal = x5.length();
+    return lengthVal;
+}
+
+function lengthOfTuple() returns (int) {
+    (int, string) a = (10, "John");
+    int lengthVal = a.length();
+    return lengthVal;
+}
+
+function lengthOfTable() returns (int) {
+    table<Employee> tbEmployee = table {
+        { primarykey id, name, salary },
```

Comment:
This should be "key"

Reply:
Renamed 'primaryKey' to 'key' with d548812